docs/controllers/scaling.md

```diff
@@ -0,0 +1,115 @@
+
+!!! warning "Scaling replicas"
+
+    It not recommended to set `replicas: 2` for an [[application]] running a normal `Controller` without leaders/shards, as this will cause both controller pods to reconcile the same objects, creating duplicate work and potential race conditions.
+
+<!-- TODO: what rollout settings do we recommend for 1 replica controllers to avoid race conditions?
+apparently you cannot set both maxSurge: 0 and maxUnavailable: 0 - https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#rolling-update-deployment
+-->
+
+The natural expiration of `leases` means that you are required to periodically update them while your main pod (the leader) is active. When your pod is to be replaced, you can initiate a step down (and expire the lease), say when you receive a `SIGTERM`. If your pod crashes, then the lease will expire naturally (albeit likely more slowly).
+
+!!! warning "Leader election is not a requirement for scaling"
+
+    When running the default 1 replica controller you have a de-facto `leader for life`. Unless you have strong latency/consistency requirements, leader election is not a required solution.
+
+
+#### Third Party Crates
+
+At the moment, leader election support is not supported by `kube` itself, and requires 3rd party crates (see [kube#485](https://github.com/kube-rs/kube/issues/485#issuecomment-1837386565)). A brief list of popular crates:
+
+- [`kube-leader-election`](https://crates.io/crates/kube-leader-election/) via [hendrikmaus](https://github.com/hendrikmaus/kube-leader-election) ([examples](https://github.com/hendrikmaus/kube-leader-election/tree/master/examples) / [docs](https://docs.rs/kube-leader-election/) / [disclaimer](https://github.com/hendrikmaus/kube-leader-election?tab=readme-ov-file#kubernetes-lease-locking))
+- [`kube-coordinate`](https://crates.io/crates/kube-coordinate) via [thedodd](https://github.com/thedodd/kube-coordinate) ([docs](https://docs.rs/kube-coordinate/))
+- [`kubert`](https://crates.io/crates/kubert) -> [`kubert::lease`](https://docs.rs/kubert/latest/kubert/lease/index.html) via [olix0r](https://github.com/olix0r/kubert) ([example](https://github.com/olix0r/kubert/blob/main/examples/lease.rs) / [linkerd use](https://github.com/linkerd/linkerd2/blob/1f4f4d417c6d06c3bd5a372fc75064f967117886/policy-controller/src/main.rs))
+
+<!-- OTHER ALTERNATIVES???
+Know other alternatives? Feel free to raise a PR here with a new list entry.
+-->
+
+### Sharding
+
+If you are unable to meet latency/resource requirements using techniques above, you may need to consider **partitioning/sharding** your resources. Below are two commonly seen approaches for sharding:
+
```

Comment:
Would we consider agents/daemons as a form of sharding? Perhaps we could offer that as an alternative for horizontally scaling a controller without requiring a lease? I'd imagine people can deploy a per-node controller responsible for reconciling state _actively assigned_ to that node. Although I think this might be quite niche since only pods ever get scheduled on a specific node. 🤷🏻  trying to think of some things we could expand on.

Reply:
ah, daemonset is a good sharding example, ill add that in!

Reply:
https://github.com/kube-rs/website/pull/58/commits/bda37fec996fcdc61653f1026cbec09246c698ae